param.go

```diff
@@ -288,22 +293,33 @@ func newParamObject(t reflect.Type) (paramObject, error) {
 				f.Name, f.Type, t)
 		}
 
-		p, err := newParam(f.Type)
-		if err != nil {
-			return po, errWrapf(err, "bad field %q of %v", f.Name, t)
-		}
+		var p param
+		if group := f.Tag.Get(_groupTag); group != "" {
+			// TODO(abg): Verify that a name or optional was not specified.
+			if f.Type.Kind() != reflect.Slice {
+				return po, fmt.Errorf("value groups may be consumed as slices only: "+
+					"field %q (%T) of %v is not a slice", f.Name, f.Type, t)
+			}
+			p = newParamGroupedSlice(group, f.Type)
+		} else {
+			var err error
+			p, err = newParam(f.Type)
+			if err != nil {
+				return po, errWrapf(err, "bad field %q of %v", f.Name, t)
```

Comment:
Why is errWrapf used here but fmt.Errorf used above?

Reply:
errWrapf is used only in places where there's an `err` to wrap. It relies on some machinery defined in error.go to make sure we track the root cause and expose it via `dig.RootCause`.